jupyterlab/labbox_ephys_widgets_jp/src/pluginWidgets.tsx

```diff
@@ -217,11 +220,39 @@ const PluginComponentWrapper: FunctionComponent<PluginComponentWrapperProps> = (
     recording = example.recording
   }
 
-  const [curation, curationDispatch] = useReducer(sortingCurationReducer, defaultSortingCuration)
-  sorting.curation = curation
+  // const [curation, curationDispatch] = useReducer(sortingCurationReducer, defaultSortingCuration)
 
-  const [selection, selectionDispatch] = useReducer(sortingSelectionReducer, defaultSortingSelection)
+  // curation
+  const [curation, curationDispatch] = useReducer(sortingCurationReducer, model.get('curation').labelChoices ? model.get('curation') : defaultSortingCuration)
```

Comment:
This code block is used to synchronize the curation between the jupyter system and the react state system.

Reply:
From Slack discussion, this is creating two hooks. One is the React `useEffect` hook which fires on changes to the curation object (or if the actual `model` reference were to change, which won't happen). This hook checks if the `model` object (a feature of the underlying Jupyter labextension framework) differs from the UI selection state. If there is a difference, the hook updates the `model` state to match the current UI state.

The second hook is attached to the `model` itself. On a change to its `curation` property, it calls the `curationDispatch` reducer with a new verb, `SetCuration`, which simply overwrites the existing curation value in the React state with whatever is in the `model`.

It is anticipated that changes would not be initiated in the `model` object, so effectively this just does a full circle to make sure everything is in sync any time a UI element changes. The conditional in line 228 (`if (model.get('curation') !== curation)`) ensures that there won't be any cyclic updates.

Reply:
yes that's right